libgnucash/app-utils/gnc-quotes.hpp

```diff
@@ -0,0 +1,148 @@
+/********************************************************************\
+ * gnc-quotes.hpp -- proxy for Finance::Quote                       *
+ * Copyright (C) 2021 Geert Janssens <geert@kobaltwit.be>           *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+\********************************************************************/
+#ifndef GNC_QUOTES_HPP
+#define GNC_QUOTES_HPP
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <gnc-commodity.hpp>  // For CommVec alias
+#include <glib.h>
+#include <stdexcept>
+
+extern  "C" {
+#include <qofbook.h>
+}
+
+using StrVec = std::vector<std::string>;
+using QuoteSources = StrVec;
+
+enum class GncQuoteError
+{
+    SUCCESS,
+    NO_RESULT,
+    QUOTE_FAILED,
+    NO_CURRENCY,
+    UNKNOWN_CURRENCY,
+    NO_PRICE,
+    UNKNOWN_PRICE_TYPE,
+    PRICE_PARSE_FAILURE,
+};
+
+/** QuoteFailure elements are namespace, mnemonic, error code, and
+ * F::Q errormsg if there is one.
+ */
+using QuoteFailure = std::tuple<std::string, std::string,
+                                GncQuoteError, std::string>;
+using QFVec = std::vector<QuoteFailure>;
+
+struct GncQuoteException : public std::runtime_error
+{
+    GncQuoteException(const std::string& msg) : std::runtime_error(msg) {}
+};
+
+const std::string not_found = std::string ("Not Found");
```

Comment:
Does this need to be marked for translation ?

Reply:
Yes.

Reply:
Actually, no, it's unneeded. `GncQuotes::GncQuotes()` will throw if Finance::Quote isn't installed and in that case there's no object to call `version()` on.

But in the course of figuring that out I also discovered that I'd neglected to change `version()` when I extracted `GncQuoteSource` and it would invariably return that "Not Found". That's fixed now.